### PR TITLE
[AIRFLOW-3528] handle socket exception with SFTPOperator

### PR DESCRIPTION
**Pull request by eladkal**
### Jira
- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW-3528) 

### Description
add try/exception to catch socket exceptions

### CODE REVIEW

**Comment by eladkal** [comment]
Can one of the committers please restart the last test?

File: airflow/contrib/operators/sftp_operator.py [hunk 1]
```diff
@@ -122,7 +122,11 @@ def execute(self, context):
                 self.ssh_hook.remote_host = self.remote_host
 
             with self.ssh_hook.get_conn() as ssh_client:
-                sftp_client = ssh_client.open_sftp()
+                try:
```
Comment (feluelle) [inline airflow/contrib/operators/sftp_operator.py:125]: Does it need to be a Custom (Airflow) Exception? 
Why can't the original exception just be raised?

Comment (eladkal) [reply airflow/contrib/operators/sftp_operator.py:125]: Currently it raises AirflowException as it's being handled at this line:
https://github.com/apache/incubator-airflow/blob/c83a4e5b952baa8d390e94d27fd12df888d3113e/airflow/contrib/operators/sftp_operator.py#L158

The propose here is just not to show the "Error while transferring (file_msg)" as this is confusing and file_msg is None.

**Comment by ashb** [comment]
This fix I don't think will do what we want, as I think now this will end up ultimately logging this as a message:

> Error while transferring None, error: Error while connecting client, error: socket timed out

(I took a guess at what might be in the OSError).

Which means we probably need to add some tests, via mocking, that makes `open_sftp` fail with an exception. And then once we have tests we can be sure the error is reported right in this case.

**Comment by eladkal** [comment]
> This fix I don't think will do what we want, as I think now this will end up ultimately logging this as a message:
> 
> > Error while transferring None, error: Error while connecting client, error: socket timed out
> 
> (I took a guess at what might be in the OSError).
> 
> Which means we probably need to add some tests, via mocking, that makes `open_sftp` fail with an exception. And then once we have tests we can be sure the error is reported right in this case.

I'm not sure what you mean... Since Python 3.3 socket.error is deprecated. It raised OSError instead. 
https://docs.python.org/3/library/socket.html#socket.error

This PR doesn't attempt to solve the problem regarding why the connection fails. It only fix the message in order to avoid confusion. This problem was reported on Slack. @XD-DENG  said that he will try to investigate about why the connection fails.

**Comment by ashb** [comment]
I guessed at what error it might report. The main thing is you have try-inside-try, so this won't do what you expect.

Additionally the error that was reporting was `'NoneType' object has no attribute 'open_sftp'` which means no exception is being thrown, instead `self.ssh_hook.get_conn() as ssh_client` is returning None.

So the except you added (capturing `OSError`) won't ever get triggered.

Tests :)

**Comment by XD-DENG** [comment]
Agree with @ashb on the exception type to catch. Based on the exception/log shared in the Slack thread, it's an `AttributeError` instead of `OSError`. This PR can not (completely) handle the issue shared in the Slack thread.

**Comment by eladkal** [comment]
OK. Got it.
In this case I'll leave this to @XD-DENG or whoever wants to pick this up. From what was reported regardless of the error handling the operator doesn't seem to work. This exceed from the scope of what I can help with at this point.

**Comment by ashb** [comment]
Okay, thanks for trying though!

**Pull request closed without merging**